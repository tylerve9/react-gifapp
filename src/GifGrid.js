import React from 'react';
import { useFecthGifs } from './hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category ).then( setImages );
    // }, [ category ]);

    // <div>
    //             { images.map( {id, title} => ( 
    //                 <p key={ img.id } > { img.title } </p> 
    //             )) }
    //         </div>

    const { data: images, loading } = useFecthGifs(category);


    return (
        <div>
            <h3 className="animate__animated animate__fadeInUpBig "> {category} </h3>
            {loading &&
                <p className=" animate__animated animate__fadeInUpBig">loading...</p>}
            {<div className="card-grid">
                {images.map(img => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>}
        </div>
    );

}

export default GifGrid;