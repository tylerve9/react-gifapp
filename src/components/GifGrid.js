import React from 'react';
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category ).then( setImages );
    // }, [ category ]);

    // <div>
    //             { images.map( {id, title} => ( 
    //                 <p key={ img.id } > { img.title } </p> 
    //             )) }
    //         </div>

    const { data: images, loading } = useFetchGifs(category);


    return (
        <div>
            <h3 className="animate__animated animate__fadeInUpBig "> {category} </h3>
            {loading &&
                <p className=" animate__animated animate__fadeInUpBig">loading...</p>}
            {<div className="card-grid">
                {images.map(img => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>}
        </div>
    );

}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}

export default GifGrid;