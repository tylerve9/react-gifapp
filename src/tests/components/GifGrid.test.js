import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre el componente <GifGrid /> ', () => {
    const category = ''
    //const wrapper = shallow( <GifGrid category={ category } /> );
    
    test('Debe renderizar el componente <GifGrid /> correctamente ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar data unavez cargada', () => {

        useFetchGifs.mockReturnValue({
                data: [{
                    id: 'ABC',
                    url: 'https://localhost/algo-jpg',
                    title: 'Cualquier cosa'
                }],
                loading: false
        });
        const wrapper = shallow( <GifGrid category={ category } /> );
        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(1);

    });
    
    
});