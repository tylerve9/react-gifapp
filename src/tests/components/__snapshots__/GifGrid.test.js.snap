// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid />  Debe renderizar el componente <GifGrid /> correctamente  1`] = `
<div>
  <h3
    className="animate__animated animate__fadeInUpBig "
  >
     
     
  </h3>
  <p
    className=" animate__animated animate__fadeInUpBig"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
